x-pack/plugins/security_solution_serverless/server/common/services/usage_reporting_service.ts

```diff
@@ -19,11 +19,13 @@ export class UsageReportingService {
     records: UsageRecord[],
     url = USAGE_SERVICE_USAGE_URL
   ): Promise<Response> {
+    const isHttps = url.includes('https');
+
     return fetch(url, {
       method: 'post',
       body: JSON.stringify(records),
       headers: { 'Content-Type': 'application/json' },
-      agent,
+      agent: isHttps ? agent : undefined, // Conditionally add agent if URL is HTTPS for supporting integration tests.
```

Comment:
cc: @joeypoon

Reply:
I guess the reason is that the `agent` attribute is not supported in the core fetch. @joeypoon, please correct me if I'm wrong.